cmd/detect.go

```diff
@@ -37,6 +38,10 @@ var detectCmd = &cobra.Command{
 			},
 		}
 
+		ui.Warning("Fan PWM association is guessed based on matching fan and pwm channel numbers. You may need to override in the config. A single pwm channel can control multiple fans.")
+		ui.Info("The device index corresponds to the hwmon channel and may not be sequential.")
+		ui.Printfln("")
```

Comment:
I don't think its a good idea to print this with every detect command.
This should be part of the documentation (the `README.md`) instead.

I also don't really understand what this is trying to say.
What is a "hwmon channel"? Is this documented somewhere?
How does it work differently from the current numbering scheme or how does it relate to it?

Reply:
> I also don't really understand what this is trying to say.
> What is a "hwmon channel"? Is this documented somewhere?
> How does it work differently from the current numbering scheme or how does it relate to it?

A channel is indeed a concept of the Linux hwmon subsystem. See https://docs.kernel.org/hwmon/hwmon-kernel-api.html and https://docs.kernel.org/hwmon/sysfs-interface.html. The high-level concept is that a "chip" (or controller) managed by the hwmon subsystem may have more than one sensor, and each such sensor will be "connected" to the chip on a particular channel. Channels are simply numbered 1...n. The sensors don't have to be the same kind -- a particular chip may have a temperature sensor on channel 1, a fan on channel 2, and a pwm on channel 3.

Now imagine I'm the happy owner of a server and it comes with some fans and I configure fan2go. And then I get unhappy with the temperatures I'm getting and since my fancy server box has a couple of unused fan slots I add a new fan, which I connect to an unused fan connector on the motherboard. This new fan will show up as "fanN" on the controller, where N corresponds to the free connector I used on the motherboard. This new fan may end on a _lowered numbered channel_ than the existing fans -- this entirely depends on the hwmon driver and which connector I used.

So a fan2go config using an enumeration index is going to break in the face of such a change, but will continue working as intended (modulo not handling the new fan) when using channel numbers.